test/tests/functional/pbs_job_default_group.py

```diff
@@ -0,0 +1,77 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2022 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of both the OpenPBS software ("OpenPBS")
+# and the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# OpenPBS is free software. You can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+#
+# OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# PBS Pro is commercially licensed software that shares a common core with
+# the OpenPBS software.  For a copy of the commercial license terms and
+# conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+# Altair Legal Department.
+#
+# Altair's dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of OpenPBS and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair's trademarks, including but not limited to "PBS™",
+# "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+# subject to Altair's trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestJobDefaultGroup(TestFunctional):
+    """
+    This test suite contains test for job which gets default group
+    """
+
+    def test_job_with_default_group(self):
+        """
+        When a user group is not present on server machine
+        then job will get a "-default-" group set on it.
+        And MOM should be able to run that job
+        """
+        if self.server.hostname == self.mom.hostname:
+            self.skipTest("Server and Execution host must be different")
+        attr = {'flatuid': True}
+        self.server.manager(MGR_CMD_SET, SERVER, attr)
+        # add a temporary new user on execution host
+        cmd = "useradd -m testpbsuser1"
+        res = self.du.run_cmd(self.mom.hostname, cmd=cmd, sudo=True)
```

Comment:
This test could actually fail in the middle, leaving the user account 'testbsuser1' in place. Then future execution of this test would fail since the user already exist.

Reply:
I have updated it to generate a random string for username with 'temppbs' as prefix.

Reply:
That's not really how you should fix this issue.  The way to handle cleaning up things which revert_to_defaults() doesn't revert is to create a class variable which is set to false.  During the test, set the class variable to true when you make the change.  In tearDown(), check the class variable, and if it is true, unset what you need to unset and set the class variable to false.

```
class TestSomething() {
    bool createdUser = false
    def test_create_user(self):
        self.createdUser = true
        createUser()

    def tearDown(self):
        if self.createdUser:
            deleteUser()
            self.createdUser = false
}
```
That's what you do.  This way if your test fails in the middle, you still delete the user.  If you do it the way you are suggesting, you'll still leave the user around as cruft.  It won't bother the next time you run the test, but you might end up leaving many test users around if the test fails many times in the middle.